perun/collect/complexity/systemtap.py

```diff
@@ -0,0 +1,320 @@
+        matching_record = trace_stack.pop()
+        return {'amount': int(record.timestamp) - int(matching_record.timestamp),
+                'uid': matching_record.name,
+                'type': 'mixed',
+                'subtype': 'time delta',
+                'structure-unit-size': matching_record.sequence}, trace_stack
+    raise exceptions.TraceStackException(record, trace_stack)
+
+
+def _process_static_record(record, trace_stack, sequence_map, probes):
+    """Processes the static output record and tries to pair it with stack record if possible
+
+    :param namedtuple record: the _TraceRecord namedtuple with parsed line values
+    :param dict trace_stack: the dictionary containing trace stack (list) for each static probe
+    :param dict sequence_map: stores the sequence counter for every static probe
+    :param list probes: the list of all static probe definitions for pairing
+    :returns tuple: the resource dictionary or empty dict, updated trace stack
+    """
+    matching_record = None
+    if record.type == RecordType.StaticSingle:
+        # The probe is paired with itself, find the last record in the stack
+        if trace_stack[record.name]:
+            matching_record = trace_stack[record.name].pop()
+            # Add the record into the trace stack to correctly measure time between each two hits
+            trace_stack[record.name].append(record._replace(sequence=sequence_map[record.name]))
+            sequence_map[record.name] += 1
+    elif record.type == RecordType.StaticEnd:
+        # Static end probe, find the starting probe record
+        name = None
+        for probe in probes:
+            if probe['pair'] == record.name:
+                name = probe['name']
+        # The stack is empty
+        if not trace_stack[name]:
+            raise exceptions.TraceStackException(record, trace_stack[name])
+        matching_record = trace_stack[name].pop()
+
+    if matching_record:
+        # Matching record was found, create resource
+        return {'amount': int(record.timestamp) - int(matching_record.timestamp),
+                'uid': record.name + '#' + matching_record.name,
+                'type': 'mixed',
+                'subtype': 'time delta',
+                'structure-unit-size': record.sequence}, trace_stack
+    else:
+        # No matching record found, insert into the stack
+        trace_stack[record.name].append(record._replace(sequence=sequence_map[record.name]))
+        sequence_map[record.name] += 1
+        return {}, trace_stack
+
+
+def _parse_record(line):
+    """ Parses line from collector output into record tuple consisting of:
+        record type, call stack offset, rule name, timestamp and sequence.
+
+    :param str line: one line from the collection output
+    :returns namedtuple: the _TraceRecord tuple
+    """
+
+    # Split the line into = 'type' 'timestamp process' : 'offset' 'rule'
+    parts = line.partition(':')
```

Comment:
Extracting the tuple returned by line.partition might be more readable, i.e. timestamp, _, right_section = line.partition(":"). Or something like that.